### ISSUE TITLE
Java debugger not working properly in wsl remote extension 

### ISSUE THREAD

**Issue by KatanaFluorescent** [microsoft/vscode-java-debug#1101]
Hello, I'm double posting this from VSCode remote extension GitHub, because I don't really where it belongs between there and here. Anyway, here is my issue :

VSCode Version: 1.63.1
Local OS Version: Win11 21H2 (build 22000.348)
Remote OS Version: 0.63.11
Remote Extension/Connection Type: WSL
Steps to Reproduce:

1)Launch a VSCode WSL remote session (here WSL2 with Ubuntu)
Write a .java file (saved in WSL, for instance un under ~) with a main method where you call a method that is defined in your JRE library, not in your project (for instance call the Exception() constructor)
2)Put a breakpoint at the line where you call this method defined outside your project
3)Run in debugger mode, and try to step into this method defined outside
4)You see the call stack evolving as it is supposed to do when you click on 'step into', but VSCode won't open the .class files where the 'defined in standard library' methods are defined (while it does open them in Windows without remote connection for instance), hence you're not able to track the execution steps of your program

Does this issue occur when you try this locally?: No

**Comment by testforstephen** [comment]
> 4)You see the call stack evolving as it is supposed to do when you click on 'step into', but VSCode won't open the .class files where the 'defined in standard library' methods are defined (while it does open them in Windows without remote connection for instance), hence you're not able to track the execution steps of your program

Usually when VS Code won't open the .class from JRE library, it means Java extension doesn't detect the src.zip for JRE library.

Could you check your JRE installation and see if src.zip is included in the location like <JAVA_HOME>/src.zip or <JAVA_HOME>/lib/src.zip?

**Comment by KatanaFluorescent** [comment]
Hello, thanks a lot for answering. It seems you are right, I check and here is what I got : 

`turbobato@15ACH6H:/mnt/c/Users/gjaze$ cd $JAVA_HOME
turbobato@15ACH6H:/usr/lib/jvm/java-17-openjdk-amd64$ ls -la
total 40
drwxr-xr-x  9 root root 4096 Dec 15 12:47 .
drwxr-xr-x  4 root root 4096 Dec 15 12:47 ..
drwxr-xr-x  2 root root 4096 Dec 15 12:47 bin
drwxr-xr-x  4 root root 4096 Dec 15 12:47 conf
lrwxrwxrwx  1 root root   42 Oct 29 11:50 docs -> ../../../share/doc/openjdk-17-jre-headless
drwxr-xr-x  3 root root 4096 Dec 15 12:47 include
drwxr-xr-x  2 root root 4096 Dec 15 12:47 jmods
drwxr-xr-x 72 root root 4096 Dec 15 12:47 legal
drwxr-xr-x  5 root root 4096 Dec 15 12:47 lib
drwxr-xr-x  3 root root 4096 Dec 15 12:47 man
-rw-r--r--  1 root root 1186 Oct 29 11:50 release`

Then : 

`turbobato@15ACH6H:/usr/lib/jvm/java-17-openjdk-amd64/lib$ ls -la | grep src.zip
lrwxrwxrwx 1 root root        24 Oct 29 11:50 src.zip -> ../../openjdk-17/src.zip`

Looks like the installation is broken and there is a symlink pointing to nothing (appearing in red in my terminal).

I installed the package openjdk-17-jdk running `sudo apt install openjdk-17-jdk`.

Here is what I get when I search for openjdk-17 in apt : 

`turbobato@15ACH6H:/usr/lib/jvm/java-17-openjdk-amd64/lib$ apt search openjdk-17
Sorting... Done
Full Text Search... Done
openjdk-17-dbg/focal-security 17.0.1+12-1~20.04 amd64
  Java runtime based on OpenJDK (debugging symbols)

openjdk-17-demo/focal-security 17.0.1+12-1~20.04 amd64
  Java runtime based on OpenJDK (demos and examples)

openjdk-17-doc/focal-security 17.0.1+12-1~20.04 all
  OpenJDK Development Kit (JDK) documentation

openjdk-17-jdk/focal-security,now 17.0.1+12-1~20.04 amd64 [installed]
  OpenJDK Development Kit (JDK)

openjdk-17-jdk-headless/focal-security,now 17.0.1+12-1~20.04 amd64 [installed,automatic]
  OpenJDK Development Kit (JDK) (headless)

openjdk-17-jre/focal-security,now 17.0.1+12-1~20.04 amd64 [installed,automatic]
  OpenJDK Java runtime, using Hotspot JIT

openjdk-17-jre-headless/focal-security,now 17.0.1+12-1~20.04 amd64 [installed,automatic]
  OpenJDK Java runtime, using Hotspot JIT (headless)

openjdk-17-jre-zero/focal-security 17.0.1+12-1~20.04 amd64
  Alternative JVM for OpenJDK, using Zero

openjdk-17-source/focal-security 17.0.1+12-1~20.04 all
  OpenJDK Development Kit (JDK) source files`
  
  Should I install the openjdk-17-source for it to work ?
